src/vs/workbench/common/editor/filteredEditorGroup.ts

```diff
@@ -0,0 +1,140 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IUntypedEditorInput, IMatchEditorOptions, EditorsOrder } from 'vs/workbench/common/editor';
+import { EditorInput } from 'vs/workbench/common/editor/editorInput';
+import { Emitter } from 'vs/base/common/event';
+import { firstOrDefault } from 'vs/base/common/arrays';
+import { IGroupModelChangeEvent, IReadonlyEditorGroupModel } from 'vs/workbench/common/editor/editorGroupModel';
+import { Disposable } from 'vs/base/common/lifecycle';
+
+abstract class FilteredEditorGroupModel extends Disposable implements IReadonlyEditorGroupModel {
+
+	private readonly _onDidModelChange = this._register(new Emitter<IGroupModelChangeEvent>());
+	readonly onDidModelChange = this._onDidModelChange.event;
+
+	constructor(
+		protected readonly model: IReadonlyEditorGroupModel
+	) {
+		super();
+
+		this._register(this.model.onDidModelChange(e => {
+			if (e.editor && this.contains(e.editor)) {
+				this._onDidModelChange.fire(e);
+			}
+		}));
+	}
```

Comment:
There are model changes that should not be filtered like that, namely all events that are on the group and not on an editor so I would always let those through.

In addition, can we have your test suite also test the filtered events?